plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiReadOptimizedDirectoryLister.java

```diff
@@ -71,16 +82,15 @@ public HudiReadOptimizedDirectoryLister(
     }
 
     @Override
-    public List<HudiFileStatus> listStatus(HudiPartitionInfo partitionInfo)
+    public List<FileSlice> listStatus(HudiPartitionInfo partitionInfo, String commitTime)
     {
+        String partition = partitionInfo.getRelativePartitionPath();
         return fileSystemView.getLatestBaseFiles(partitionInfo.getRelativePartitionPath())
-                .map(HudiBaseFile::getFileEntry)
-                .map(fileEntry -> new HudiFileStatus(
-                        fileEntry.location(),
-                        false,
-                        fileEntry.length(),
-                        fileEntry.lastModified().toEpochMilli(),
-                        max(blockSize(fileEntry.blocks()), min(fileEntry.length(), MIN_BLOCK_SIZE))))
+                .map(baseFile -> new FileSlice(
```

Comment:
we have to be mindful of RO query with partition path updates and the partitions not compacted yet. RO can show dups. Can take it up as a followup. Not a blocker for this PR.

Reply:
Yes.  I think that's not the problem of the connector implementation but rather how RO query works.